src/lib.rs

```diff
@@ -428,6 +428,24 @@ impl Build {
         self
     }
 
+    /// removes a flag from the existing flags.
+    /// example:
+    /// ```no_run
+    /// cc::Build::new()
+    ///     .file("src/foo.c")
+    ///     .flag("-fake_flag")
+    ///     .remove_flag("-fake_flag")
+    ///     .compile("foo");
+    /// ```
+    pub fn remove_flag(&mut self, flag: &'static str) -> &mut Self{
+        let flag: Arc<str> = flag.into();
+        if self.flags.contains(&flag){
+            self.flags.retain(|other_flag| other_flag != &flag);
+        }
```

Comment:
You don't have to convert it into an `Arc` (which creates a heap allocation) and calling `contains` and `retain` causes two iteration to be made:

```suggestion
        self.flags.retain(|other_flag| &**other_flag != flag);
```